### ISSUE TITLE
[Bug] Active Recordings table fails to update when restarting a recording with `replace: ALWAYS`

### ISSUE THREAD

**Issue by aali309** [cryostatio/cryostat-web#1286]
### Current Behavior

![Screenshot from 2024-06-14 13-06-42](https://github.com/cryostatio/cryostat-web/assets/56743004/cf2bf1ea-2bf5-4d6b-b03c-85e25d4ff814)

### Expected Behavior

Active Recordings should update to reflect the create recording notification.

### Steps To Reproduce

1. checkout on cryostatio/cryostat main branch
2.  build and run smoketest
3. visit `localhost:8080` 
4. navigate to `Recordings`  table
5. run the below query on terminal and see the behavior on the web UI

``` 
`http --follow -v --auth='user:pass' POST :8080/api/v3/graphql \
      Content-Type:application/json \
      query='
  query {
    targetNodes(filter: { annotations: ["PORT = 9091"] }) {
      name
      target {
        doStartRecording(recording: {
          name: "test",
          template: "Profiling",
          templateType: "TARGET",
          replace: "ALWAYS"
        }) {
          name
          state
        }
      }
    }
  }'

```

### Environment

```Markdown
- OS: 
- Environment: 
- Version:
```

**Comment by andrewazores** [comment]
This is just a notification handling issue, right? If I understand correctly, the notification appears but the table does not update. If you navigate away and back, or refresh, the table contains the correct new content?

**Comment by andrewazores** [comment]
Please also check for any other similar cases with this view/table not updating. It sounds like the particular query you used to perform the update would not be the only way to trigger this behaviour. Like, is the `replace: ALWAYS` the only time it breaks, or do `STOPPED` and `NEVER` do it too?

**Comment by tthvo** [comment]
I did a quick test locally and the behaviour is correct with the notification updating the table. Might just be another issue with flanky notification handling.

![image](https://github.com/cryostatio/cryostat-web/assets/68053619/ae06a3e6-2c54-477b-bc82-59b5a7daec74)

**Comment by aali309** [comment]
> This is just a notification handling issue, right? If I understand correctly, the notification appears but the table does not update. If you navigate away and back, or refresh, the table contains the correct new content?

Yes, just like what was happening with the Archived tables before and labels.

**Comment by aali309** [comment]
> Please also check for any other similar cases with this view/table not updating. It sounds like the particular query you used to perform the update would not be the only way to trigger this behaviour. Like, is the `replace: ALWAYS` the only time it breaks, or do `STOPPED` and `NEVER` do it too?

This is my next test. I will be looking for those and similar like 'restart: true'. I saw Stopping a running recoding updates as expected.

**Comment by tthvo** [comment]
@aali309 how is the testing this issue? Is this persistent and only for `replace: ALWAYS`?

**Comment by aali309** [comment]
> @aali309 how is the testing this issue? Is this persistent and only for `replace: ALWAYS`?

So far, for `replace:NEVER` works as expected for all scenarios and recordings.

`replace:ALWAYS` and 
`replace:STOPPED`  on the already exist recordings (i.e `startup` and `onstart`) works as expected. The table updates and the right notification is emitted.
![image](https://github.com/cryostatio/cryostat-web/assets/56743004/2b23e5e4-112d-4556-a3a6-7adc51758f33)

However, on newly created recordings (i.e test, test2 etc), the table does not update until page is refreshed or navigating away and back, and different notification is emitted compared to the above. (So only newly created recordings have this issue but not the existing ones (i.e `startup` and `onstart`).
The query and response works well the only issue is on the UI (different notification and updating the table).
```
❯ http --follow -v --auth='user:pass' POST :8080/api/v3/graphql \
      Content-Type:application/json \
      query='
  query {
    targetNodes(filter: { annotations: ["PORT = 9091"] }) {
      name
      target {
        doStartRecording(recording: {
          name: "test",
          template: "Profiling",
          templateType: "TARGET",
          replace: "STOPPED" # you can replace this with "ALWAYS"
        }) {
          name
          state
        }
      }
    }
  }'

POST /api/v3/graphql HTTP/1.1
Accept: application/json, */*;q=0.5
Accept-Encoding: gzip, deflate
Authorization: Basic dXNlcjpwYXNz
Connection: keep-alive
Content-Length: 333
Content-Type: application/json
Host: localhost:8080
User-Agent: HTTPie/3.2.2

{
    "query": "\nquery {\n  targetNodes(filter: { annotations: [\"PORT = 9091\"] }) {\n    name\n    target {\n      doStartRecording(recording: {\n        name: \"test\",\n        template: \"Profiling\",\n        templateType: \"TARGET\",\n        replace: \"STOPPED\"\n      }) {\n        name\n        state\n      }\n    }\n  }\n}"
}


HTTP/1.1 200 OK
Content-Length: 153
Content-Type: application/json
Date: Tue, 18 Jun 2024 14:07:37 GMT
Gap-Auth: user

{
    "data": {
        "targetNodes": [
            {
                "name": "service:jmx:rmi:///jndi/rmi://cryostat3:9091/jmxrmi",
                "target": {
                    "doStartRecording": {
                        "name": "test",
                        "state": "RUNNING"
                    }
                }
            }
        ]
    }
}

```

![image](https://github.com/cryostatio/cryostat-web/assets/56743004/f38ba7d5-39ae-4f9d-b562-8c8d5f514b4a)

**Comment by Josh-Matsuoka** [comment]
Creating a new recording with replace:ALWAYS seems to work fine for me, however calling the query multiple times results in some strange behavior. The recording disappears from the table, with notifications listed in the order:

Recording created 
Recording deleted

However, refreshing or navigating away and back shows the recording is there.

This makes sense from the table's POV, it got a notification that the recording was created and deleted immediately, and acted accordingly, though the overall state is incorrect.

It seems like a problem in the ordering in which notifications are being passed to the client.

**Comment by Josh-Matsuoka** [comment]
Looking further into the server side this behavior is also consistent, on subsequent calls of the query with some logging added for when notifications are fired we get this:

`[Hibernate] 
    insert 
    into
        ActiveRecording
        (continuous, duration, external, maxAge, maxSize, metadata, name, remoteId, startTime, state, target_id, toDisk, id) 
    values
        (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)

2025-01-17 16:16:40,029 WARN  [io.cry.rec.ActiveRecording$Listener] (vert.x-worker-thread-1) ========== PostPersist Called ============
[Hibernate] 
    delete 
    from
        ActiveRecording 
    where
        id=?

2025-01-17 16:16:40,031 WARN  [io.cry.rec.ActiveRecording$Listener] (vert.x-worker-thread-1) ============= PostRemove Called =================`

The PostPersist hook creates the Recording created notification while the PostRemove hook creates the Recording deleted notification. The web client gets these in order and removes the recording from the table.

calling refreshRecordingsList in the notification handler instead of using the notification fields to update the table fixes this and makes the table work as expected, though this results in more calls to the api (getTargetRecordings), I think it's the only way to really keep the state accurate.
